src/main/java/ch/njol/skript/conditions/CondIsWearing.java

```diff
@@ -57,32 +45,36 @@ public class CondIsWearing extends Condition {
 	
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] vars, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+	public boolean init(Expression<?>[] vars, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		entities = (Expression<LivingEntity>) vars[0];
 		types = (Expression<ItemType>) vars[1];
 		setNegated(matchedPattern == 1);
 		return true;
 	}
 	
 	@Override
-	public boolean check(final Event e) {
-		return entities.check(e,
-				en -> types.check(e,
-						t -> {
-							EntityEquipment equip = en.getEquipment();
+	public boolean check(Event event) {
+		return entities.check(event,
+				entity -> types.check(event,
+						type -> {
+							EntityEquipment equip = entity.getEquipment();
 							if (equip == null)
 								return false; // No equipment -> not wearing anything
-							for (final ItemStack is : equip.getArmorContents()) {
-								if (t.isOfType(is) ^ t.isAll())
-									return !t.isAll();
+
+							ItemStack[] contents = Arrays.copyOf(equip.getArmorContents(), equip.getArmorContents().length + 1);
+							contents[contents.length - 1] = equip.getItem(EquipmentSlot.BODY);
+
+							for (ItemStack content : contents) {
+								if (type.isOfType(content) ^ type.isAll())
+									return !type.isAll();
 							}
-							return t.isAll();
+							return type.isAll();
```

Comment:
This could/should be simplified with `EquipmentSlot.values()` no? It seems weird do this